src/HealthChecks.SmbCifs/SmbCifsStorageHealthCheck.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using SharpCifs.Smb;
+
+namespace HealthChecks.SmbCifs
+{
+    public class SmbCifsStorageHealthCheck : IHealthCheck
+    {
+        private readonly object _cifsOptions;
+
+        private NtlmPasswordAuthentication _auth;
+
+        public SmbCifsStorageHealthCheck(SmbCifsBasicOptions cifsOptions)
+        {
+            _cifsOptions = cifsOptions ?? throw new ArgumentNullException(nameof(cifsOptions));
+
+            if (_cifsOptions.GetType() != typeof(SmbCifsBasicOptions))
+                throw new InvalidCastException($"Invalid cast for object {nameof(cifsOptions)} to SmbCifsBasicOptions");
+        }
+
+        public SmbCifsStorageHealthCheck(SmbCifsExtendedOptions cifsOptions)
+        {
+            _cifsOptions = cifsOptions ?? throw new ArgumentNullException(nameof(cifsOptions));
+
+            if (_cifsOptions.GetType() != typeof(SmbCifsExtendedOptions))
+                throw new InvalidCastException($"Invalid cast for object {nameof(cifsOptions)} to SmbCifsExtendedOptions");
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(
+          HealthCheckContext context,
+          CancellationToken cancellationToken = default)
+        {
+            var hostName = _cifsOptions.GetType() == typeof(SmbCifsBasicOptions)
```

Comment:
Maybe type inheritance?

Reply:
Sorry don't quite understand ,. U the way happy new year ! 🥳

Reply:
What did you want to achieve with such a strict check on the type of object?